nmap.yaml

```diff
@@ -0,0 +1,37 @@
+package:
+  name: nmap
+  version: 7.94
+  epoch: 0
+  description: "network discovery and security auditing tool"
+  copyright:
+    - license: GPL-2.0
+
+environment:
+  contents:
+    packages:
+      - wolfi-base
+      - build-base
+      - openssl-dev
+
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://nmap.org/dist/nmap-${{package.version}}.tar.bz2
+      expected-sha256: d71be189eec43d7e099bac8571509d316c4577ca79491832ac3e1217bc8f92cc
+
+  - uses: autoconf/configure
+
+  - uses: autoconf/make
+
+  - uses: autoconf/make-install
+
+  - uses: strip
+
+  - runs: |
+      # nmap require cap_net_raw,cap_net_admin,cap_net_bind_service+eip while working nonroot
+      # capabilities can define but they don't pass to container.
```

Comment:
Sure, I will check it.

Reply:
Still not working. Running with --prvileged flag like that `nmap -v --privileged -A scanme.nmap.org` I'm still facing `Couldn't open a raw socket. Error: Operation not permitted (1)`

![image](https://github.com/wolfi-dev/os/assets/30419196/0ade4ff1-027e-44c1-ad1e-38a51597f59b)

**Melange file:**

![image](https://github.com/wolfi-dev/os/assets/30419196/b2bd073f-affe-4f27-911e-bf44439ebce2)

Reply:
Is there any way to verify capabilities in runtime? @kaniini 
I tried `getcap` in container but  I couldn't see any cap 

![image](https://github.com/wolfi-dev/os/assets/30419196/b77377ea-c62f-461c-a6dc-f603df2e7ee1)

Reply:
@ahmetak4n I was able to run with the setcap (with a non root user ignore the name) 

```
bash-5.2$ getcap /usr/bin/nmap
/usr/bin/nmap cap_net_bind_service,cap_net_raw=ep
bash-5.2$ id
uid=65532(spark-operator) gid=65532(spark-operator) groups=65532(spark-operator)
bash-5.2$ nmap -v --privileged -A scanme.nmap.org
Starting Nmap 7.94 ( https://nmap.org ) at 2023-06-19 05:39 UTC
NSE: Loaded 156 scripts for scanning.
NSE: Script Pre-scanning.
Initiating NSE at 05:39
Completed NSE at 05:39, 0.00s elapsed
Initiating NSE at 05:39
Completed NSE at 05:39, 0.00s elapsed
Initiating NSE at 05:39
Completed NSE at 05:39, 0.00s elapsed
Initiating Ping Scan at 05:39
Scanning scanme.nmap.org (45.33.32.156) [4 ports]
Completed Ping Scan at 05:39, 0.01s elapsed (1 total hosts)
Initiating Parallel DNS resolution of 1 host. at 05:39
Completed Parallel DNS resolution of 1 host. at 05:39, 0.00s elapsed
Initiating SYN Stealth Scan at 05:39
Scanning scanme.nmap.org (45.33.32.156) [1000 ports]
Discovered open port 22/tcp on 45.33.32.156
Discovered open port 80/tcp on 45.33.32.156
Discovered open port 9929/tcp on 45.33.32.156
Discovered open port 31337/tcp on 45.33.32.156
Completed SYN Stealth Scan at 05:39, 1.56s elapsed (1000 total ports)
Initiating Service scan at 05:39
Scanning 4 services on scanme.nmap.org (45.33.32.156)
Completed Service scan at 05:40, 6.05s elapsed (4 services on 1 host)
Initiating OS detection (try #1) against scanme.nmap.org (45.33.32.156)
Retrying OS detection (try #2) against scanme.nmap.org (45.33.32.156)
Initiating Traceroute at 05:40
Completed Traceroute at 05:40, 0.02s elapsed
Initiating Parallel DNS resolution of 2 hosts. at 05:40
Completed Parallel DNS resolution of 2 hosts. at 05:40, 0.07s elapsed
NSE: Script scanning 45.33.32.156.
Initiating NSE at 05:40
Completed NSE at 05:40, 5.16s elapsed
Initiating NSE at 05:40
Completed NSE at 05:40, 0.10s elapsed
Initiating NSE at 05:40
Completed NSE at 05:40, 0.00s elapsed
Nmap scan report for scanme.nmap.org (45.33.32.156)
Host is up (0.0058s latency).
Other addresses for scanme.nmap.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
Not shown: 992 closed tcp ports (reset)
PORT      STATE    SERVICE      VERSION
22/tcp    open     ssh          OpenSSH 6.6.1p1 Ubuntu 2ubuntu2.13 (Ubuntu Linux; protocol 2.0)
| ssh-hostkey: 
|   1024 ac:00:a0:1a:82:ff:cc:55:99:dc:67:2b:34:97:6b:75 (DSA)
|   2048 20:3d:2d:44:62:2a:b0:5a:9d:b5:b3:05:14:c2:a6:b2 (RSA)
|_  256 33:fa:91:0f:e0:e1:7b:1f:6d:05:a2:b0:f1:54:41:56 (ED25519)
25/tcp    filtered smtp
80/tcp    open     http         Apache httpd 2.4.7 ((Ubuntu))
| http-methods: 
|_  Supported Methods: GET HEAD POST OPTIONS
|_http-title: Go ahead and ScanMe!
|_http-favicon: Nmap Project
|_http-server-header: Apache/2.4.7 (Ubuntu)
135/tcp   filtered msrpc
139/tcp   filtered netbios-ssn
445/tcp   filtered microsoft-ds
9929/tcp  open     nping-echo   Nping echo
31337/tcp open     tcpwrapped
Device type: printer|WAP
Running (JUST GUESSING): HP embedded (86%), Netgear embedded (85%)
OS CPE: cpe:/h:hp:officejet_pro_8500 cpe:/h:netgear:wgr614v7
Aggressive OS guesses: HP Officejet Pro 8500 printer (86%), HP PSC 2400-series Photosmart printer (85%), Netgear WGR614v7 wireless broadband router (85%)
No exact OS matches for host (test conditions non-ideal).
Uptime guess: 34.372 days (since Mon May 15 20:43:51 2023)
Network Distance: 3 hops
TCP Sequence Prediction: Difficulty=264 (Good luck!)
IP ID Sequence Generation: Incremental
Service Info: OS: Linux; CPE: cpe:/o:linux:linux_kernel

TRACEROUTE (using port 80/tcp)
HOP RTT     ADDRESS
1   0.10 ms 172.17.0.1
2   0.14 ms 192.168.65.5
3   0.48 ms scanme.nmap.org (45.33.32.156)

NSE: Script Post-scanning.
Initiating NSE at 05:40
Completed NSE at 05:40, 0.00s elapsed
Initiating NSE at 05:40
Completed NSE at 05:40, 0.00s elapsed
Initiating NSE at 05:40
Completed NSE at 05:40, 0.00s elapsed
Read data files from: /usr/bin/../share/nmap
OS and Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
Nmap done: 1 IP address (1 host up) scanned in 17.04 seconds
           Raw packets sent: 1080 (51.500KB) | Rcvd: 1029 (41.756KB)
           ```
           
    you can try the same container here ```docker run -it ttl.sh/nmap9```( images expires in a day)

Reply:
Thanks @ajayk. Is everything the same as above configuration? I wonder why I can't repeat it 🤔

Reply:
@ahmetak4n it was a long weekend couldn't respond earlier